File: prometheo/cgen/code.c
void p_class_init(p_class *object){
    object->attr_1 = 1;
    object->attr_2 = 3.0;
    object->_Z8method_1type_1type_2 = &_Z8method_1type_1type_2_impl;
    object->_Z8method_1type_1 = &_Z8method_1type_1_impl;
    object->_Z8method_2type_1 = &_Z8method_2type_1_impl;
    object->_Z11loop_methodtype_1 = &_Z11loop_methodtype_1_impl;
}


double _Z8method_1type_1type_2_impl(p_class *self, type_1 arg1, type_2 arg2) {
    double a = arg1.field1;
    double b = arg2.field2;
    double copyarg2 = arg2;
    double c = a * b + b * a * a;
    return c;
}


double _Z8method_1type_1_impl(p_class *self, type_1 arg1) {
    double a = arg1.field1;
    double b = arg1.field2;
    double copyarg1 = arg1;
    double c = a * b + b * a * a;
    return c;
}


double _Z8method_2type_1_impl(p_class *self, type_1 arg1) {
    double c = arg1 * arg1;
    return c;
}


int _Z11loop_methodtype_1_impl(p_class *self, type_1 arg1) {
    int res = 0;
    double var1 = 0;
    var1 = self.method_2(arg1);
    for (int i = 0; i < arg1; i++);
        res = res + 1;
    return res;
}

File: prometeo/cgen/code.py
class p_class:
    attr_1: int = 1
    attr_2: double = 3.0

    def method_1(arg1: type_1, arg2: type_2) -> double:
        a: double = arg1.field1
        b: double = arg2.field2
        copyarg2: double = arg2
        c: double = a*b + b*a*a

        return c

    def method_1(arg1: type_1) -> double:
        a: double = arg1.field1
        b: double = arg1.field2
        copyarg1: double = arg1
        c: double = a*b + b*a*a

        return c
    
    def method_2(arg1: type_1) -> double:
        c: double = arg1*arg1
        return c

    def loop_method(arg1: type_1) -> int:
        res: int = 0
        var1: double = 0
        var1 = self.method_2(arg1)
        
        for i in range(arg1):
            res = res + 1
        return res
